src/app/app.component.ts

```diff
@@ -11,23 +11,26 @@ import { environment } from '../environments/environment';
 export class AppComponent implements OnInit {
   numValidators: number;
   inProgress = false;
-  depositContractAddress = environment.depositContractAddress;
-  
+  depositContractAddress: string;
+
   constructor(
     private readonly web3: NoAccessWeb3Service,
+    private readonly contractService: ContractService,
     private readonly progress: ProgressService,
   ) {
     this.progress.progress.subscribe(v => this.inProgress = v);
   }
 
-  async ngOnInit() {
-    await this.updateValidatorCount();
-
-    this.web3.depositEvents().on('data', () => this.updateValidatorCount());
+  ngOnInit() {
+    this.contractService.getAddress().subscribe(async (res: any) => {
```

Comment:
Please do not use `any`. `any` is your enemy and it takes all of the benefits away from typescript. It _must_ be a string if you are assigning it to depositContractAddress. Please set it to string.